File: lib/widgets/faculty_details_card.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import '../models/faculty_model.dart';

class FacultyDetailsCard extends StatefulWidget {
  final Faculty? faculty;

  const FacultyDetailsCard({super.key, required this.faculty});

  @override
  State<FacultyDetailsCard> createState() => _FacultyDetailsCardState();
}

class _FacultyDetailsCardState extends State<FacultyDetailsCard>
    with TickerProviderStateMixin {
  late AnimationController _animationController;
  late Animation<double> _slideAnimation;
  late Animation<double> _fadeAnimation;

  @override
  void initState() {
    super.initState();
    _animationController = AnimationController(
      duration: const Duration(milliseconds: 600),
      vsync: this,
    );
    _slideAnimation = Tween<double>(begin: 0.3, end: 1.0).animate(
      CurvedAnimation(parent: _animationController, curve: Curves.easeOutCubic),
    );
    _fadeAnimation = Tween<double>(begin: 0.0, end: 1.0).animate(
      CurvedAnimation(parent: _animationController, curve: Curves.easeInOut),
    );
    _animationController.forward();
  }

  @override
  void dispose() {
    _animationController.dispose();
    super.dispose();
  }

  void _copyToClipboard(String text, String label) {
    Clipboard.setData(ClipboardData(text: text));
    HapticFeedback.lightImpact();
    ScaffoldMessenger.of(context).showSnackBar(
      SnackBar(
        content: Row(
          children: [
            const Icon(Icons.content_copy_rounded, color: Colors.white, size: 20),
            const SizedBox(width: 12),
            Text('$label copied to clipboard',
                style: const TextStyle(fontWeight: FontWeight.w600)),
          ],
        ),
        backgroundColor: const Color(0xFFE83F25),
        behavior: SnackBarBehavior.floating,
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
        margin: const EdgeInsets.all(16),
        elevation: 8,
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    if (widget.faculty == null) {
      return Scaffold(
        appBar: AppBar(
          title: const Text('Faculty Details'),
          backgroundColor: const Color(0xFFA62C2C),
        ),
        body: const Center(
          child: Text('No faculty data available'),
        ),
      );
    }

    final faculty = widget.faculty!;

    return Scaffold(
      backgroundColor: const Color(0xFFFFF8F5),
      appBar: AppBar(
        elevation: 0,
        backgroundColor: Colors.transparent,
        foregroundColor: const Color(0xFFA62C2C),
        title: const Text(
          'Faculty Profile',
          style: TextStyle(
            fontWeight: FontWeight.w800,
            fontSize: 22,
            color: Color(0xFFA62C2C),
          ),
        ),
        centerTitle: false,
        leading: Container(
          margin: const EdgeInsets.all(8),
          decoration: BoxDecoration(
            color: const Color(0xFFA62C2C).withOpacity(0.1),
            borderRadius: BorderRadius.circular(12),
          ),
          child: IconButton(
            icon: const Icon(Icons.arrow_back_rounded, color: Color(0xFFA62C2C)),
            onPressed: () => Navigator.pop(context),
          ),
        ),
      ),
      body: FadeTransition(
        opacity: _fadeAnimation,
        child: ScaleTransition(
          scale: _slideAnimation,
          child: SingleChildScrollView(
            physics: const BouncingScrollPhysics(),
            padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 8),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                // Hero Profile Section
                Container(
                  width: double.infinity,
                  margin: const EdgeInsets.only(bottom: 24),
                  decoration: BoxDecoration(
                    gradient: const LinearGradient(
                      colors: [Color(0xFFA62C2C), Color(0xFFE83F25)],
                      begin: Alignment.topLeft,
                      end: Alignment.bottomRight,
                    ),
                    borderRadius: BorderRadius.circular(24),
                    boxShadow: [
                      BoxShadow(
                        color: const Color(0xFFA62C2C).withOpacity(0.3),
                        blurRadius: 20,
                        offset: const Offset(0, 8),
                      ),
                    ],
                  ),
                  child: Stack(
                    children: [
                      Positioned(
                        top: -50,
                        right: -50,
                        child: Container(
                          width: 120,
                          height: 120,
                          decoration: BoxDecoration(
                            color: Colors.white.withOpacity(0.1),
                            shape: BoxShape.circle,
                          ),
                        ),
                      ),
                      Padding(
                        padding: const EdgeInsets.all(28),
                        child: Column(
                          crossAxisAlignment: CrossAxisAlignment.start,
                          children: [
                            Row(
                              children: [
                                Container(
                                  width: 70,
                                  height: 70,
                                  decoration: BoxDecoration(
                                    color: Colors.white,
                                    borderRadius: BorderRadius.circular(18),
                                    boxShadow: [
                                      BoxShadow(
                                        color: Colors.black.withOpacity(0.1),
                                        blurRadius: 8,
                                        offset: const Offset(0, 4),
                                      ),
                                    ],
                                  ),
                                  child: const Icon(
                                    Icons.person_rounded,
                                    color: Color(0xFFA62C2C),
                                    size: 35,
                                  ),
                                ),
                                const SizedBox(width: 20),
                                Expanded(
                                  child: Column(
                                    crossAxisAlignment: CrossAxisAlignment.start,
                                    children: [
                                      Text(
                                        faculty.name,
                                        style: const TextStyle(
                                          fontSize: 26,
                                          fontWeight: FontWeight.w800,
                                          color: Colors.white,
                                        ),
                                      ),
                                      const SizedBox(height: 6),
                                      Container(
                                        padding: const EdgeInsets.symmetric(
                                            horizontal: 12, vertical: 6),
                                        decoration: BoxDecoration(
                                          color: Colors.white.withOpacity(0.2),
                                          borderRadius: BorderRadius.circular(20),
                                        ),
                                        child: Text(
                                          faculty.designation,
                                          style: const TextStyle(
                                            fontSize: 12,
                                            fontWeight: FontWeight.w600,
                                            color: Colors.white,
                                          ),
                                        ),
                                      ),
                                    ],
                                  ),
                                ),
                              ],
                            ),
                            const SizedBox(height: 20),
                            Container(
                              padding: const EdgeInsets.symmetric(
                                  horizontal: 16, vertical: 10),
                              decoration: BoxDecoration(
                                color: Colors.white.withOpacity(0.15),
                                borderRadius: BorderRadius.circular(16),
                                border: Border.all(
                                  color: Colors.white.withOpacity(0.3),
                                ),
                              ),
                              child: Row(
                                children: [
                                  const Icon(
                                    Icons.business_rounded,
                                    color: Colors.white,
                                    size: 18,
                                  ),
                                  const SizedBox(width: 8),
                                  Text(
                                    faculty.department,
                                    style: const TextStyle(
                                      fontSize: 16,
                                      fontWeight: FontWeight.w600,
                                      color: Colors.white,
                                    ),
                                  ),
                                ],
                              ),
                            ),
                          ],
                        ),
                      ),
                    ],
                  ),
                ),

                // Contact Section
                _buildModernSection(
                  title: 'Contact',
                  icon: Icons.phone_rounded,
                  color: const Color(0xFFEA7300),
                  children: [
                    _buildContactItem(
                      'Email',
                      faculty.email,
                      Icons.mail_outline_rounded,
                      onTap: () => _copyToClipboard(faculty.email, 'Email'),
                    ),
                    const SizedBox(height: 12),
                    _buildContactItem(
                      'Phone',
                      faculty.contact,
                      Icons.phone_outlined,
                      onTap: () => _copyToClipboard(faculty.contact, 'Phone'),
                    ),
                  ],
                ),

                const SizedBox(height: 20),

                // Personal Info Section
                _buildModernSection(
                  title: 'Personal Information',
                  icon: Icons.person_outline_rounded,
                  color: const Color(0xFFD3CA79),
                  children: [
                    _buildInfoRow('Date of Birth', faculty.dateOfBirth, Icons.cake_outlined),
                    const SizedBox(height: 16),
                    _buildInfoRow('Address', faculty.address, Icons.location_on_outlined,
                        isExpandable: true,
                        onTap: () => _copyToClipboard(faculty.address, 'Address')),
                  ],
                ),

                const SizedBox(height: 20),

                // Professional Info Section
                _buildModernSection(
                  title: 'Professional Details',
                  icon: Icons.work_outline_rounded,
                  color: const Color(0xFFE83F25),
                  children: [
                    _buildInfoRow('Department', faculty.department, Icons.apartment_rounded),
                    const SizedBox(height: 16),
                    _buildInfoRow('Position', faculty.designation, Icons.badge_outlined),
                  ],
                ),

                const SizedBox(height: 32),

                // Action Buttons
                Row(
                  children: [
                    Expanded(
                      flex: 3,
                      child: Container(
                        height: 56,
                        decoration: BoxDecoration(
                          gradient: const LinearGradient(
                            colors: [Color(0xFFEA7300), Color(0xFFD3CA79)],
                            begin: Alignment.centerLeft,
                            end: Alignment.centerRight,
                          ),
                          borderRadius: BorderRadius.circular(16),
                          boxShadow: [
                            BoxShadow(
                              color: const Color(0xFFEA7300).withOpacity(0.4),
                              blurRadius: 12,
                              offset: const Offset(0, 4),
                            ),
                          ],
                        ),
                        child: ElevatedButton.icon(
                          onPressed: () {
                            HapticFeedback.lightImpact();
                            ScaffoldMessenger.of(context).showSnackBar(
                              const SnackBar(
                                content: Text('Edit functionality to be implemented'),
                              ),
                            );
                          },
                          style: ElevatedButton.styleFrom(
                            backgroundColor: Colors.transparent,
                            shadowColor: Colors.transparent,
                            shape: RoundedRectangleBorder(
                              borderRadius: BorderRadius.circular(16),
                            ),
                          ),
                          icon: const Icon(Icons.edit_rounded,
                              color: Colors.white, size: 22),
                          label: const Text(
                            'Edit Profile',
                            style: TextStyle(
                              color: Colors.white,
                              fontWeight: FontWeight.w700,
                              fontSize: 16,
                            ),
                          ),
                        ),
                      ),
                    ),
                    const SizedBox(width: 16),
                    Container(
                      height: 56,
                      width: 56,
                      decoration: BoxDecoration(
                        color: const Color(0xFFA62C2C).withOpacity(0.1),
                        borderRadius: BorderRadius.circular(16),
                        border: Border.all(
                          color: const Color(0xFFA62C2C).withOpacity(0.3),
                          width: 2,
                        ),
                      ),
                      child: IconButton(
                        onPressed: () {
                          HapticFeedback.lightImpact();
                          _showDeleteDialog();
                        },
                        icon: const Icon(
                          Icons.delete_outline_rounded,
                          color: Color(0xFFA62C2C),
                          size: 24,
                        ),
                      ),
                    ),
                  ],
                ),

                const SizedBox(height: 20),
              ],
            ),
          ),
        ),
      ),
    );
  }

  Widget _buildModernSection({
    required String title,
    required IconData icon,
    required Color color,
    required List<Widget> children,
  }) {
    return Container(
      width: double.infinity,
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(20),
        boxShadow: [
          BoxShadow(
            color: Colors.black.withOpacity(0.08),
            blurRadius: 16,
            offset: const Offset(0, 4),
          ),
        ],
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Container(
            width: double.infinity,
            padding: const EdgeInsets.all(20),
            decoration: BoxDecoration(
              color: color.withOpacity(0.1),
              borderRadius: const BorderRadius.only(
                topLeft: Radius.circular(20),
                topRight: Radius.circular(20),
              ),
            ),
            child: Row(
              children: [
                Container(
                  padding: const EdgeInsets.all(10),
                  decoration: BoxDecoration(
                    color: color,
                    borderRadius: BorderRadius.circular(12),
                  ),
                  child: Icon(
                    icon,
                    color: Colors.white,
                    size: 20,
                  ),
                ),
                const SizedBox(width: 16),
                Text(
                  title,
                  style: TextStyle(
                    fontSize: 18,
                    fontWeight: FontWeight.w800,
                    color: color,
                  ),
                ),
              ],
            ),
          ),
          Padding(
            padding: const EdgeInsets.all(20),
            child: Column(children: children),
          ),
        ],
      ),
    );
  }

  Widget _buildContactItem(String label, String value, IconData icon, {VoidCallback? onTap}) {
    return InkWell(
      onTap: onTap,
      borderRadius: BorderRadius.circular(14),
      child: Container(
        padding: const EdgeInsets.all(16),
        decoration: BoxDecoration(
          color: const Color(0xFFF8F9FA),
          borderRadius: BorderRadius.circular(14),
          border: Border.all(color: const Color(0xFFE9ECEF)),
        ),
        child: Row(
          children: [
            Container(
              padding: const EdgeInsets.all(8),
              decoration: BoxDecoration(
                color: const Color(0xFFEA7300).withOpacity(0.1),
                borderRadius: BorderRadius.circular(8),
              ),
              child: Icon(icon, color: const Color(0xFFEA7300), size: 18),
            ),
            const SizedBox(width: 16),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    label,
                    style: const TextStyle(
                      fontSize: 12,
                      fontWeight: FontWeight.w600,
                      color: Color(0xFF6C757D),
                    ),
                  ),
                  const SizedBox(height: 2),
                  Text(
                    value.isEmpty ? 'Not provided' : value,
                    style: TextStyle(
                      fontSize: 16,
                      fontWeight: FontWeight.w600,
                      color: value.isEmpty
                          ? const Color(0xFFADB5BD)
                          : const Color(0xFF212529),
                    ),
                  ),
                ],
              ),
            ),
            if (onTap != null)
              const Icon(
                Icons.content_copy_rounded,
                size: 18,
                color: Color(0xFFADB5BD),
              ),
          ],
        ),
      ),
    );
  }

  Widget _buildInfoRow(String label, String value, IconData icon,
      {bool isExpandable = false, VoidCallback? onTap}) {
    return InkWell(
      onTap: onTap,
      borderRadius: BorderRadius.circular(12),
      child: Container(
        padding: const EdgeInsets.symmetric(vertical: 8),
        child: Row(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Icon(icon, color: const Color(0xFF6C757D), size: 20),
            const SizedBox(width: 16),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    label,
                    style: const TextStyle(
                      fontSize: 12,
                      fontWeight: FontWeight.w600,
                      color: Color(0xFF6C757D),
                    ),
                  ),
                  const SizedBox(height: 4),
                  Text(
                    value.isEmpty ? 'Not provided' : value,
                    style: TextStyle(
                      fontSize: 16,
                      fontWeight: FontWeight.w600,
                      color: value.isEmpty
                          ? const Color(0xFFADB5BD)
                          : const Color(0xFF212529),
                    ),
                    maxLines: isExpandable ? 3 : 1,
                    overflow: TextOverflow.ellipsis,
                  ),
                ],
              ),
            ),
            if (onTap != null)
              const Icon(
                Icons.content_copy_rounded,
                size: 16,
                color: Color(0xFFADB5BD),
              ),
          ],
        ),
      ),
    );
  }

  void _showDeleteDialog() {
    showDialog(
      context: context,
      builder: (ctx) => AlertDialog(
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(24)),
        backgroundColor: Colors.white,
        title: Row(
          children: [
            Container(
              padding: const EdgeInsets.all(12),
              decoration: BoxDecoration(
                color: const Color(0xFFA62C2C).withOpacity(0.1),
                borderRadius: BorderRadius.circular(16),
              ),
              child: const Icon(Icons.warning_amber_rounded,
                  color: Color(0xFFA62C2C), size: 28),
            ),
            const SizedBox(width: 16),
            const Text('Delete Faculty',
                style: TextStyle(
                    fontSize: 20,
                    fontWeight: FontWeight.w800,
                    color: Color(0xFF212529))),
          ],
        ),
        content: const Text(
          'Are you sure you want to delete this faculty? This action cannot be undone.',
          style: TextStyle(fontSize: 16, color: Color(0xFF6C757D)),
        ),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(ctx),
            style: TextButton.styleFrom(
              padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 14),
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(14),
              ),
            ),
            child: const Text('Cancel',
                style: TextStyle(
                    color: Color(0xFF6C757D), fontWeight: FontWeight.w600, fontSize: 16)),
          ),
          Container(
            decoration: BoxDecoration(
              gradient: const LinearGradient(
                colors: [Color(0xFFA62C2C), Color(0xFFE83F25)],
              ),
              borderRadius: BorderRadius.circular(14),
            ),
            child: ElevatedButton(
              onPressed: () {
                Navigator.pop(ctx);
                ScaffoldMessenger.of(context).showSnackBar(
                  const SnackBar(
                    content: Text('Delete functionality to be implemented'),
                  ),
                );
              },
              style: ElevatedButton.styleFrom(
                backgroundColor: Colors.transparent,
                shadowColor: Colors.transparent,
                padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 14),
                shape: RoundedRectangleBorder(
                  borderRadius: BorderRadius.circular(14),
                ),
              ),
              child: const Text('Delete',
                  style: TextStyle(
                      fontWeight: FontWeight.w700, color: Colors.white, fontSize: 16)),
            ),
          ),
        ],
      ),
    );
  }
}

File: lib/widgets/FacultyDetailsCard.java
import React, { useState, useEffect } from 'react';
import { ArrowLeft, Mail, Phone, Calendar, MapPin, Edit, Trash2, Copy, Check, User, Briefcase } from 'lucide-react';

        const FacultyDetailsCard = () => {
        const [animationStarted, setAnimationStarted] = useState(false);
  const [copiedItem, setCopiedItem] = useState('');
  const [showDeleteDialog, setShowDeleteDialog] = useState(false);

// Sample faculty data
  const faculty = {
name: "Dr. Sarah Johnson",
designation: "Associate Professor",
department: "Computer Science & Engineering",
email: "sarah.johnson@university.edu",
contact: "+1 (555) 123-4567",
dateOfBirth: "March 15, 1985",
address: "123 University Avenue, Academic City, AC 12345"
        };

useEffect(() => {
setAnimationStarted(true);
  }, []);

          const copyToClipboard = (text, label) => {
        navigator.clipboard.writeText(text);
setCopiedItem(label);
setTimeout(() => setCopiedItem(''), 2000);
        };

        const ContactCard = ({ icon: Icon, label, value, color, onCopy }) => (
    <div
onClick={() => onCopy && onCopy()}
className={`p-4 bg-white rounded-2xl border-2 cursor-pointer transition-all duration-300 hover:shadow-lg ${
color === 'red' ? 'border-red-100 hover:border-red-200' : 'border-orange-100 hover:border-orange-200'
        }`}
style={{
boxShadow: color === 'red'
        ? '0 2px 8px rgba(166, 44, 44, 0.1)'
        : '0 2px 8px rgba(232, 63, 37, 0.1)'
        }}
        >
      <div className="flex items-center">
        <div className={`p-2.5 rounded-xl ${
color === 'red' ? 'bg-red-50' : 'bg-orange-50'
        }`}>
          <Icon className={`w-5 h-5 ${
    color === 'red' ? 'text-red-600' : 'text-orange-600'
}`} />
        </div>
        <div className="ml-4 flex-1">
          <p className="text-xs font-semibold text-gray-500 uppercase tracking-wide">
        {label}
          </p>
          <p className="text-base font-bold text-gray-800 mt-1">
        {value || 'Not provided'}
          </p>
        </div>
        {onCopy && (
          <div className={`p-1.5 rounded-lg ${
color === 'red' ? 'bg-red-50' : 'bg-orange-50'
        }`}>
        {copiedItem === label ? (
              <Check className={`w-4 h-4 ${
    color === 'red' ? 'text-red-600' : 'text-orange-600'
}`} />
        ) : (
              <Copy className={`w-4 h-4 ${
    color === 'red' ? 'text-red-600' : 'text-orange-600'
}`} />
        )}
          </div>
        )}
      </div>
    </div>
        );

        const InfoTile = ({ icon: Icon, label, value, color, onCopy }) => (
    <div
onClick={() => onCopy && onCopy()}
className={`p-4 bg-white rounded-2xl border border-gray-200 transition-all duration-300 hover:shadow-md ${
onCopy ? 'cursor-pointer' : ''
        }`}
        >
      <div className="flex items-center">
        <div
className="p-2.5 rounded-xl"
style={{ backgroundColor: `${color}25` }}
        >
          <Icon className="w-5 h-5" style={{ color }} />
        </div>
        <div className="ml-4 flex-1">
          <p className="text-xs font-semibold text-gray-500 uppercase tracking-wide">
        {label}
          </p>
          <p className="text-base font-bold text-gray-800 mt-1">
        {value || 'Not provided'}
          </p>
        </div>
        {onCopy && (
          <div
className="p-1.5 rounded-lg"
style={{ backgroundColor: `${color}15` }}
        >
        {copiedItem === label ? (
              <Check className="w-4 h-4" style={{ color }} />
        ) : (
              <Copy className="w-4 h-4" style={{ color }} />
        )}
          </div>
        )}
      </div>
    </div>
        );

        const SectionHeader = ({ title, icon: Icon }) => (
    <div className="flex items-center mb-3">
      <div className="p-2 bg-gradient-to-r from-red-600 to-orange-500 rounded-xl">
        <Icon className="w-5 h-5 text-white" />
      </div>
      <h2 className="ml-3 text-xl font-extrabold text-gray-800 tracking-wide">
        {title}
      </h2>
    </div>
        );

        const DeleteDialog = () => (
    <div className="fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center p-4 z-50">
      <div className="bg-white rounded-3xl p-6 max-w-md w-full shadow-2xl">
        <div className="text-center">
          <div className="p-4 bg-red-50 rounded-full inline-block mb-4">
            <Trash2 className="w-8 h-8 text-red-600" />
          </div>
          <h3 className="text-xl font-extrabold text-gray-800 mb-2">
Delete Faculty
          </h3>
          <p className="text-gray-600 mb-6 leading-relaxed">
Are you sure you want to delete this faculty member? This action cannot be undone and all data will be permanently removed.
          </p>
          <div className="flex space-x-3">
            <button
onClick={() => setShowDeleteDialog(false)}
className="flex-1 py-4 px-6 border-2 border-gray-300 text-gray-700 font-semibold rounded-xl hover:bg-gray-50 transition-colors"
        >
Cancel
        </button>
            <button
onClick={() => {
setShowDeleteDialog(false);
alert('Delete functionality to be implemented');
              }}
className="flex-1 py-4 px-6 bg-red-600 text-white font-bold rounded-xl hover:bg-red-700 transition-colors"
        >
Delete
        </button>
          </div>
        </div>
      </div>
    </div>
        );

        return (
    <div className="min-h-screen bg-orange-50">
        {/* Header with Hero Section */}
      <div
className={`relative bg-gradient-to-b from-red-600 via-red-500 to-orange-500 transition-all duration-700 ease-out ${
animationStarted ? 'opacity-100 scale-100' : 'opacity-0 scale-95'
        }`}
style={{ height: '280px' }}
        >
        {/* Decorative circles */}
        <div className="absolute -top-12 -right-12 w-48 h-48 bg-white bg-opacity-10 rounded-full"></div>
        <div className="absolute -bottom-20 -left-20 w-40 h-40 bg-white bg-opacity-8 rounded-full"></div>

        {/* Navigation */}
        <div className="absolute top-4 left-4 z-10">
          <button className="p-2 bg-white bg-opacity-20 rounded-xl backdrop-blur-sm">
            <ArrowLeft className="w-6 h-6 text-white" />
          </button>
        </div>

        {/* Profile Content */}
        <div className="absolute bottom-10 left-0 right-0 text-center px-6">
          <div className="w-24 h-24 bg-white rounded-3xl mx-auto mb-4 flex items-center justify-center shadow-2xl">
            <User className="w-12 h-12 text-orange-500" />
          </div>
          <h1 className="text-3xl font-extrabold text-white mb-2 tracking-wide">
        {faculty.name}
          </h1>
          <div className="inline-block px-5 py-2 bg-white bg-opacity-20 backdrop-blur-sm rounded-full border border-white border-opacity-30">
            <p className="text-sm font-semibold text-white tracking-wide">
        {faculty.designation}
            </p>
          </div>
        </div>
      </div>

        {/* Content */}
      <div className="p-6 -mt-4 relative z-10">
        {/* Department Card */}
        <div className="bg-white p-5 rounded-2xl shadow-lg mb-5">
          <div className="flex items-center">
            <div className="p-3 bg-gradient-to-r from-orange-500 to-yellow-500 rounded-2xl">
              <Briefcase className="w-6 h-6 text-white" />
            </div>
            <div className="ml-4">
              <p className="text-xs font-medium text-gray-500 uppercase tracking-wide">
Department
        </p>
              <p className="text-lg font-bold text-gray-800 mt-1">
        {faculty.department}
              </p>
            </div>
          </div>
        </div>

        {/* Contact Section */}
        <SectionHeader title="Contact Details" icon={Phone} />
        <div className="space-y-3 mb-6">
          <ContactCard
icon={Mail}
label="Email"
value={faculty.email}
color="red"
onCopy={() => copyToClipboard(faculty.email, 'Email')}
        />
          <ContactCard
icon={Phone}
label="Phone"
value={faculty.contact}
color="orange"
onCopy={() => copyToClipboard(faculty.contact, 'Phone')}
        />
        </div>

        {/* Personal Information */}
        <SectionHeader title="Personal Information" icon={User} />
        <div className="space-y-3 mb-8">
          <InfoTile
icon={Calendar}
label="Date of Birth"
value={faculty.dateOfBirth}
color="#EA7300"
        />
          <InfoTile
icon={MapPin}
label="Address"
value={faculty.address}
color="#D3CA79"
onCopy={() => copyToClipboard(faculty.address, 'Address')}
        />
        </div>

        {/* Action Buttons */}
        <div className="flex space-x-4">
          <button
onClick={() => alert('Edit functionality to be implemented')}
className="flex-1 h-14 bg-gradient-to-r from-red-600 to-red-500 text-white font-bold rounded-2xl flex items-center justify-center shadow-lg hover:shadow-xl transition-all duration-300"
        >
            <Edit className="w-5 h-5 mr-2" />
Edit Profile
          </button>
          <button
onClick={() => setShowDeleteDialog(true)}
className="w-14 h-14 bg-white border-2 border-red-200 text-red-600 rounded-2xl flex items-center justify-center shadow-lg hover:shadow-xl transition-all duration-300"
        >
            <Trash2 className="w-6 h-6" />
          </button>
        </div>
      </div>

        {/* Copy Success Notification */}
        {copiedItem && (
        <div className="fixed bottom-6 left-6 right-6 bg-red-600 text-white p-4 rounded-2xl shadow-2xl flex items-center">
          <Check className="w-5 h-5 mr-3" />
          <span className="font-semibold">{copiedItem} copied successfully</span>
        </div>
        )}

        {/* Delete Dialog */}
        {showDeleteDialog && <DeleteDialog />}
    </div>
        );
        };

export default FacultyDetailsCard;